libs/langchain/langchain/text_splitter.py

```diff
@@ -391,16 +391,23 @@ def split_text(self, text: str) -> List[Document]:
         initial_metadata: Dict[str, str] = {}
 
         in_code_block = False
+        opening_fence = ""
 
         for line in lines:
             stripped_line = line.strip()
 
-            if stripped_line.startswith("```"):
-                # code block in one row
-                if stripped_line.count("```") >= 2:
+            if not in_code_block:
+                # Exclude inline code spans
+                if stripped_line.startswith("```") and stripped_line.count("```") == 1:
+                    in_code_block = True
+                    opening_fence = "```"
+                elif stripped_line.startswith("~~~"):
```

Comment:
do we need to do the same count check here , in case start and end of block are in a single line

Reply:
No, see https://spec.commonmark.org/0.30/#example-146
tl;dr: if 2 `~~~`'s appear in one line, the latter is considered part of info string, instead of end of block